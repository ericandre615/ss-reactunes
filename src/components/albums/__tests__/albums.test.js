import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AlbumList from '../list';
import AlbumCard from '../album-card';

const mockAlbums = [
  {
    collectionId: 280176511,
    artistName: 'Guided By Voices',
    collectionName: 'Propeller',
    artistViewUrl: 'https://mock.artist.url',
    collectionViewUrl: 'https://mock.collection.url/propeller',
    artworkUrl100: 'https://mock.album.art/source/100x100aa.jpg',
    copyright: '℗ 1992 Scat',
    releaseDate: '1992-02-14T08:00:00Z',
  },
  {
    collectionId: 328165532,
    artistName: 'Guided By Voices',
    collectionName: 'Suitcase - Failed Experiments and Trashed Aircraft',
    artistViewUrl: 'https://mock.artist.url',
    collectionViewUrl: 'https://mock.collection.url/suitcase',
    artworkUrl100: 'https://mock.album.art/100x100bb.png',
    copyright: '℗ 2000 Fading Captain Series',
    releaseDate: '2000-12-01T08:00:00Z',
  },
];

describe('Albums Components', () => {
  it('should match empty snapshot', () => {
    const ShallowAlbumList = shallow(<AlbumList />);

    expect(toJson(ShallowAlbumList)).toMatchSnapshot();
  });

  it('should match snapshot with list', () => {
    const ShallowAlbumList = shallow(
      <AlbumList albums={ mockAlbums } artworkSize={ 400 } />,
    );

    expect(toJson(ShallowAlbumList)).toMatchSnapshot();
  });

  it('should match single AlbumCard snapshot', () => {
    const ShallowAlbum = shallow(<AlbumCard album={ mockAlbums[0] } albumSize={ 600 } />);

    expect(toJson(ShallowAlbum)).toMatchSnapshot();
  });
});
