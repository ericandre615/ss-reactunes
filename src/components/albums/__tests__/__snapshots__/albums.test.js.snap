// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Albums Components should match empty snapshot 1`] = `
<ul
  className="albums-list"
/>
`;

exports[`Albums Components should match single AlbumCard snapshot 1`] = `
<article
  className="album-card"
>
  <div
    className="img-mask"
  >
    <img
      alt="Guided By Voices Propeller Album Cover"
      src="https://mock.album.art/source/100x100aa.jpg"
    />
  </div>
  <div
    className="album-info"
  >
    <span
      className="album-name-mask"
    >
      <h1
        className="album-name"
      >
        Propeller
      </h1>
    </span>
    <h2
      className="artist-name"
    >
      Guided By Voices
    </h2>
    <h3
      className="release-date"
    >
      February 1992
    </h3>
  </div>
</article>
`;

exports[`Albums Components should match snapshot with list 1`] = `
<ul
  className="albums-list"
>
  <li
    className="album-item"
    key="280176511"
  >
    <NavLink
      to="/albums/280176511"
    >
      <AlbumCard
        album={
          Object {
            "artistName": "Guided By Voices",
            "artistViewUrl": "https://mock.artist.url",
            "artworkUrl100": "https://mock.album.art/source/100x100aa.jpg",
            "collectionId": 280176511,
            "collectionName": "Propeller",
            "collectionViewUrl": "https://mock.collection.url/propeller",
            "copyright": "℗ 1992 Scat",
            "releaseDate": "1992-02-14T08:00:00Z",
          }
        }
        artworkSize={400}
      />
    </NavLink>
  </li>
  <li
    className="album-item"
    key="328165532"
  >
    <NavLink
      to="/albums/328165532"
    >
      <AlbumCard
        album={
          Object {
            "artistName": "Guided By Voices",
            "artistViewUrl": "https://mock.artist.url",
            "artworkUrl100": "https://mock.album.art/100x100bb.png",
            "collectionId": 328165532,
            "collectionName": "Suitcase - Failed Experiments and Trashed Aircraft",
            "collectionViewUrl": "https://mock.collection.url/suitcase",
            "copyright": "℗ 2000 Fading Captain Series",
            "releaseDate": "2000-12-01T08:00:00Z",
          }
        }
        artworkSize={400}
      />
    </NavLink>
  </li>
</ul>
`;
